pkg/ottl/ottlfuncs/README.md

```diff
@@ -589,6 +590,61 @@ Examples:
 
 - `ExtractPatterns(body, "^(?P<timestamp>\\w+ \\w+ [0-9]+:[0-9]+:[0-9]+) (?P<hostname>([A-Za-z0-9-_]+)) (?P<process>\\w+)(\\[(?P<pid>\\d+)\\])?: (?P<message>.*)$")`
 
+### ExtractGrokPatterns
+
+`ExtractGrokPatterns(target, pattern, Optional[namedCapturesOnly], Optional[patternDefinitions])`
+
+The `ExtractGrokPatterns` Converter parses unstructured data into a format that is structured and queryable. 
+It returns a `pcommon.Map` struct that is a result of extracting named capture groups from the target string. If no matches are found then an empty `pcommon.Map` is returned.
+
+- `target` is a Getter that returns a string. 
+- `pattern` is a grok pattern string. 
+- `namedCapturesOnly` (optional) specifies if non-named captures should be returned. 
+- `patternDefinitions` (optional) is a list of custom pattern definition strings used inside `pattern` in the form of `PATTERN_NAME=PATTERN`. 
+This parameter lets you define your own custom patterns to improve readability when the extracted `pattern` is not part of the default set or when you need custom naming. 
+
+If `target` is not a string or nil `ExtractGrokPatterns` returns an error. If `pattern` does not contain at least 1 named capture group and `namedCapturesOnly` is set to `true` then `ExtractPatterns` errors on startup.
+
+Parsing is done using [Elastic Go-Grok](https://github.com/elastic/go-grok?tab=readme-ov-file#default-set-of-patterns) library.
```

Comment:
```suggestion
Parsing is done using [Elastic Go-Grok](https://github.com/elastic/go-grok?tab=readme-ov-file) library.
```
Should we just link to the readme instead of the "default set of patterns" section?